services/organizer_service.js

```diff
@@ -3,36 +3,39 @@ const { Checker } = require('./data_checker')
 const { Authorize } = require('./jwt_service')
 const Organizer = models.Organizer
 const { Responses } = require('../utils/responses')
-const { Meeting } = require('../db/meeting')
 const createResponse = Responses.createResponse
 
 const invalidResponse = createResponse("Invalid login or password", 400)
 
-const findOrganizerWithEmail = async (email) => await Organizer.findAll({ where: { email: email } })
+const generateRandomString = () => Math.random().toString(36).slice(-8);
 
 const organizerService = {
     register: async (email, password) => {
         const checkResult = Checker.checkEmail(email)
         if (checkResult !== true) return createResponse(checkResult, 400)
 
-        const organizers = await findOrganizerWithEmail(email)
-        if (organizers.length !== 0) return createResponse("This mail is already used", 400)
+        const organizer = await Authorize.findOrganizerWithEmail(email)
+        if (organizer !== null) return createResponse("This mail is already used", 400)
 
         const newPassword = await Authorize.hashPassword(password)
         await Organizer.create({ email: email, password: newPassword })
         return createResponse("Created user", 201)
     },
 
     login: async (email, password) => {
-        const organizers = await findOrganizerWithEmail(email)
-        if (organizers.length === 0) return invalidResponse
+        const organizer = await Authorize.findOrganizerWithEmail(email)
+        if (organizer === null) return invalidResponse
 
-        const organizer = organizers[0]
         return await Authorize.checkPassword(password, organizer)
             ? createResponse(Authorize.generateAccessToken(organizer))
             : invalidResponse
     },
 
+    loginWithEmail: async (email) => {
+        let organizer = await Authorize.findOrganizerWithEmail(email)
+        if (organizer === null) await organizerService.register(email, generateRandomString())
```

Comment:
To jest już taki trochę komentarz do czepialstwa, ale samego tego awaita wyniósłbym do osobnej funkcji. Z tego, co rozumiem, to tu jest właśnie zakładanie konta w tym systemie osobie, która zalogowana jest w innym systemie, tak? Jeśli faktycznie tak jest, to nie powinniśmy się nad tym zastanawiać.

Reply:
Znaczy funkcja register jest async więc no await to tak z przyzwyczajenia no i że endpoint zwróci poprawnie jak w całości przejdzie proces